flutter/lib/desktop/pages/connection_page.dart

```diff
@@ -171,36 +236,143 @@ class _ConnectionPageState extends State<ConnectionPage>
-                      style: const TextStyle(
-                        fontFamily: 'WorkSans',
-                        fontSize: 22,
-                        height: 1.4,
-                      ),
-                      maxLines: 1,
-                      cursorColor:
-                          Theme.of(context).textTheme.titleLarge?.color,
-                      decoration: InputDecoration(
-                          filled: false,
-                          counterText: '',
-                          hintText: _idInputFocused.value
-                              ? null
-                              : translate('Enter Remote ID'),
-                          contentPadding: const EdgeInsets.symmetric(
-                              horizontal: 15, vertical: 13)),
-                      controller: _idController,
-                      inputFormatters: [IDTextInputFormatter()],
-                      onSubmitted: (s) {
-                        onConnect();
-                      },
-                    ),
-                  ),
+                  child: 
+                   Autocomplete<Peer>(
+                    optionsBuilder: (TextEditingValue textEditingValue) {
+                      if (textEditingValue.text == '') {
+                        return const Iterable<Peer>.empty();
+                      }
+                      else if (peers.isEmpty) {
+                         Peer emptyPeer = Peer(
+                          id: '',
+                          username: '',
+                          hostname: '',
+                          alias: '',
+                          platform: '',
+                          tags: [],
+                          hash: '',
+                          forceAlwaysRelay: false,
+                          rdpPort: '',
+                          rdpUsername: '',
+                          loginName: '',
+                        );
+                        return [emptyPeer];
+                      }
+                      else {
+                        String textWithoutSpaces = textEditingValue.text.replaceAll(" ", "");
+                        if (int.tryParse(textWithoutSpaces) != null) {
+                          textEditingValue = TextEditingValue(
+                            text: textWithoutSpaces,
+                            selection: textEditingValue.selection,
+                          );
+                        }
+
+                        return peers.where((peer) =>
+                        peer.id.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.username.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.hostname.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.alias.toLowerCase().contains(textEditingValue.text.toLowerCase()) ||
+                        peer.platform.toLowerCase().contains(textEditingValue.text.toLowerCase()))
```

Comment:
>If alias exists, should search alias only, ignore id, please confirm if the search in peer tab has this logic.
 
 Peer tab can search by id.
 Also, I think keeping alias and id both will be helpful in case user is entering the id and if it already exists.
<img width="1111" alt="Screenshot 2023-10-18 at 3 58 31 PM" src="https://github.com/rustdesk/rustdesk/assets/73148455/7694a173-9b61-4fc3-abd8-3a2daefe0f73">

Reply:
Did you show both id and alias if there are both?

Reply:
Peer card does only show alias if there is alias and id, I think you can improve here, show both alias and id as `alias / id` if there are both.

Reply:
do you mean something like this, if alias exists?
<img width="338" alt="Screenshot 2023-10-18 at 4 17 19 PM" src="https://github.com/rustdesk/rustdesk/assets/73148455/58ddfdcc-fee3-4050-867e-c1a5cf393c0e">

Reply:
test_duplicate(12341234)

(12341234) can be smaller font and ligher color.